File: study/그리디/03-숫자 카드 게임.py
import sys
getInput = sys.stdin.readline

n, m = map(int, getInput().rstrip().split())
result = 0

for i in range(n):
  data = list(map(int, getInput().rstrip().split()))
  min_value = min(data)
  result = max(result, min_value)

print(result)

File: solving/나동빈-이것이코딩테스트이다/그리디/03-숫자 카드 게임.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const n = parseInt(input[0][0]);
let result = 0;

for (let i = 1; i <= n; i++) {
  const data = input[i].split(' ').map((el) => parseInt(el));
  const min_value = Math.min.apply(null, data);
  result = Math.max(result, min_value);
}

console.log(result);
